packages/govuk-frontend/src/govuk/components/skip-link/skip-link.mjs

```diff
@@ -31,28 +32,43 @@ export class SkipLink extends GOVUKFrontendComponent {
     this.$module = $module
 
     // Check for linked element
-    const $linkedElement = this.getLinkedElement()
-    if (!$linkedElement) {
-      return
+    try {
+      const $linkedElement = this.getLinkedElement()
+      this.$linkedElement = $linkedElement
+    } catch (errorCause) {
+      const cause =
+        errorCause instanceof Error ? errorCause : new Error(String(errorCause))
+      throw new MissingElementError('The linked HTML element does not exist', {
+        cause
+      })
     }
 
-    this.$linkedElement = $linkedElement
     this.$module.addEventListener('click', () => this.focusLinkedElement())
   }
 
   /**
    * Get linked element
    *
    * @private
-   * @returns {HTMLElement | null} $linkedElement - DOM element linked to from the skip link
+   * @throws {Error} If the "href" attribute does not contain a hash
+   * @throws {Error} If the element with the specified ID does not exist
+   * @returns {HTMLElement} $linkedElement - DOM element linked to from the skip link
    */
   getLinkedElement() {
     const linkedElementId = this.getFragmentFromUrl()
     if (!linkedElementId) {
-      return null
+      throw new Error(
+        `The Skip Link's "href" attribute does not contain a hash`
+      )
+    }
+
+    const linkedElement = document.getElementById(linkedElementId)
+
+    if (!linkedElement) {
+      throw new Error(`Element with ID "${linkedElementId}" does not exist`)
```

Comment:
Might be worth making these generic?

Mainly so monitor tools that scrape errors can contain `Target selector "${XXX}" not found`

```suggestion
      throw new Error(`Skip link: Target selector "#${linkedElementId}" not found`)
```